tunnel/intercept/tproxy/tproxy_linux.go

```diff
@@ -70,26 +76,45 @@ var listenConfig = net.ListenConfig{
 	},
 }
 
-func New(lanIf string) (intercept.Interceptor, error) {
-	return NewWithDiverter(lanIf, "")
+type Config struct {
+	LanIf            string
+	Diverter         string
+	UDPIdleTimeout   time.Duration
+	UDPCheckInterval time.Duration
 }
 
-func NewWithDiverter(lanIf, diverter string) (intercept.Interceptor, error) {
+func New(config Config) (intercept.Interceptor, error) {
+	log := pfxlog.Logger()
+
 	self := &interceptor{
-		lanIf:          lanIf,
-		diverter:       diverter,
-		serviceProxies: cmap.New[*tProxy](),
-		ipt:            nil,
+		lanIf:            config.LanIf,
+		diverter:         config.Diverter,
+		udpIdleTimeout:   config.UDPIdleTimeout,
+		udpCheckInterval: config.UDPCheckInterval,
+		serviceProxies:   cmap.New[*tProxy](),
+		ipt:              nil,
 	}
 
-	if diverter != "" {
-		cmd := exec.Command(diverter, "-V")
+	if self.udpIdleTimeout < 5*time.Second {
+		self.udpIdleTimeout = DefaultUdpIdleTimeout
+	}
+	if self.udpCheckInterval < time.Second {
+		self.udpCheckInterval = DefaultUdpCheckInterval
```

Comment:
Might want to log a warning for both of the defaults being applied.

Reply:
added some logging